dbms/src/Client/Completion.cpp

```diff
@@ -0,0 +1,128 @@
+#include <readline/readline.h>
+#include <cstdlib>
+#include "Completion.h"
+
+namespace Completion
+{
+    static uint hashpjw(const char *arKey, uint nKeyLength)
+    {
+        uint h = 0;
+        uint g;
+        uint i;
+        for (i = 0; i < nKeyLength; i++) {
+            h = (h << 4) + arKey[i];
+            if ((g = (h & 0xF0000000))) {
+                h = h ^ (g >> 24);
+                h = h ^ g;
+            }
+        }
+        return h;
+    }
+
+    int init_hash_table(HashTable *ht, size_t size)
```

Comment:
From what source this code was copy-pasted?

Reply:
It is from https://github.com/mysql/mysql-server/blob/8.0/client/completion_hash.cc. I've put source in Completion.h file.